data-prepper-api/src/test/java/com/amazon/dataprepper/model/event/EventMetadataImplTest.java

```diff
@@ -0,0 +1,117 @@
+
+    @Before
+    public void setup() {
+        testAttributes = new HashMap<>();
+        testAttributes.put(UUID.randomUUID().toString(), UUID.randomUUID());
+        testAttributes.put(UUID.randomUUID().toString(), UUID.randomUUID().toString());
+
+        testTimeReceived = Instant.now();
+
+        testEventType = UUID.randomUUID().toString();
+
+        eventMetadata = new EventMetadataImpl.Builder()
+                .withAttributes(testAttributes)
+                .withEventType(testEventType)
+                .withTimeReceived(testTimeReceived)
+                .build();
+    }
+
+    @Test
+    public void testGetEventType() {
+        final String eventType = eventMetadata.getEventType();
+        assertThat(eventType, is(not(emptyOrNullString())));
+        assertThat(eventType, is(equalTo(testEventType)));
+    }
+
+    @Test
+    public void testGetTimeReceived() {
+        final Instant timeReceived = eventMetadata.getTimeReceived();
+        assertThat(timeReceived, is(notNullValue()));
+        assertThat(timeReceived, is(equalTo(testTimeReceived)));
+    }
+
+    @Test
+    public void testGetAttributes() {
+        final Map<String, Object> attributes = eventMetadata.getAttributes();
+
+        assertThat(attributes, is(not(anEmptyMap())));
+        assertThat(attributes, is(equalTo(testAttributes)));
+    }
+
+    @Test
+    public void testAttributesMutation_throwsAnException() {
+        final Map<String, Object> attributes = eventMetadata.getAttributes();
+
+        assertThrows(UnsupportedOperationException.class, () -> attributes.put("foo", "bar"));
+    }
+
+    @Test
+    public void testBuild_withoutTimeReceived() {
+
+        final Instant before = Instant.now();
+
+        EventMetadata result = new EventMetadataImpl.Builder()
+                .withEventType(testEventType)
+                .build();
+
+        assertThat(result, notNullValue());
+        final Instant timeReceived = result.getTimeReceived();
+        assertThat(timeReceived, notNullValue());
+        assertThat(timeReceived, is(greaterThan(before)));
+        assertThat(timeReceived, is(lessThan(Instant.now())));
+    }
+
+    @Test
+    public void testBuild_withoutMap() {
+
+        final EventMetadata result = new EventMetadataImpl.Builder()
+                .withEventType(testEventType)
+                .build();
+
+        assertThat(result, notNullValue());
+
+        final Map<String, Object> attributes = result.getAttributes();
+        assertThat(attributes, notNullValue());
+        assertThat(attributes, is(anEmptyMap()));
+    }
+
+    @Test
+    public void testBuild_withoutEventType_throwsAnException() {
+        assertThrows(NullPointerException.class, () -> new EventMetadataImpl.Builder().build());
+    }
+
+    @Test
+    public void testBuild_withoutEmptyEventType_throwsAnException() {
+        assertThrows(IllegalArgumentException.class, () -> new EventMetadataImpl.Builder().withEventType("").build());
```

Comment:
It is best to reduce what is in the `assertThrows` so that you are sure you are testing the correct method fails.

An exception from either `withType()` or `build()` will cause the exception. Which is it that we are validating?

You can extract the Builder into a variable and then reduce it to: `assertThrows(IllegalArgumentException.class(), builder.build());`